src/eos-crash-metrics.c

```diff
@@ -137,6 +140,70 @@ normalize_path (char *path)
     path[i] = path[i] == '!' ? '/' : path[i];
 }
 
+static GPtrArray *
+get_installed_ref_for_executable (FlatpakInstallation *installation,
+                                  const char *executable_name,
+                                  GError **error)
+{
+  g_autoptr(GPtrArray) xrefs = NULL;
+  GPtrArray *applications = NULL;
+  FlatpakInstalledRef *xref = NULL;
+  guint i;
+
+  /* get apps and runtimes */
+	xrefs = flatpak_installation_list_installed_refs (installation,
+                                                    NULL, error);
+  if (xrefs == NULL)
+    return NULL;
+
+  applications = g_ptr_array_new ();
```

Comment:
Consider using `g_ptr_array_new_full` here to pre-allocate a vector of up to `len` refs. In any event you will need to use `g_ptr_array_new_full` and pass `g_object_unref` to the `GDestroyNotify` parameter to ensure tae the `FlatpakInstalledRef` is unreffed when the returned argument is destroyed.